autoload/lsp.vim

```diff
@@ -46,6 +58,67 @@ function! lsp#log(...) abort
     endif
 endfunction
 
+let s:has_lua = has('nvim-0.4.0') || (has('lua') && has('patch-8.2.0775'))
+let s:lua_array_start_index = has('nvim-0.4.0') || has('patch-8.2.1066')
+
+function! s:init_lua() abort
+  lua <<EOF
+  function my_pairs(servers)
```

Comment:
this will make the function global.

Can we start moving lua functions to lua folder similar to this https://github.com/prabirshrestha/vim-lsp/tree/lua-lsp-client/lua/lsp.

here again diff is an exception since it was brought from other libraries and would like to make sure it is easy to update.

Also use `local M = {}` and `return M` with `local function` pattern.

Reply:
It seems that require('xxx') in every `luaeval` cost about 20% extra times in my env.